scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/policy/license/LicenseSynchronization.java

```diff
@@ -0,0 +1,218 @@
+                                                storeFile.getAbsolutePath(), e);
+            }
+        }
+    }
+
+    void addSoftware(String software, int nbTokens) {
+        persistedJobsLicenses.put(software, new LinkedBlockingQueue<String>(nbTokens));
+        persistedTasksLicenses.put(software, new LinkedBlockingQueue<String>(nbTokens));
+        remainingTokens.put(software, nbTokens);
+    }
+
+    void persist() {
+        try {
+            recordManager.commit();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    LinkedBlockingQueue<String> getPersistedJobsLicense(String license) {
+        return persistedJobsLicenses.get(license);
+    }
+
+    boolean containsJobId(String license, String jobId) {
+        return persistedJobsLicenses.get(license).stream().anyMatch(jid -> jid.equals(jobId));
+    }
+
+    void addJobToLicense(String license, String jobId) {
+        persistedJobsLicenses.get(license).add(jobId);
+        remainingTokens.put(license, remainingTokens.get(license) - 1);
+    }
+
+    void markJobLicenseChange(String license) {
+        // Record Manager mark as dirty (uncommited) entries which have be modified via a put call
+        // Thus, such operation as persistedJobsLicenses.get(license).dosomething() will not be committed
+        // by the following trick, we mark the entry as dirty and commit
+        persistedJobsLicenses.put(license, persistedJobsLicenses.get(license));
+    }
+
+    LinkedBlockingQueue<String> getPersistedTasksLicense(String license) {
+        return persistedTasksLicenses.get(license);
+    }
+
+    void addTaskToLicense(String license, String taskId) {
+        persistedTasksLicenses.get(license).add(taskId);
+        remainingTokens.put(license, remainingTokens.get(license) - 1);
+    }
+
+    void markTaskLicenseChange(String license) {
+        // Record Manager mark as dirty (uncommited) entries which have be modified via a put call
+        // Thus, such operation as persistedTasksLicenses.get(license).dosomething() will not be committed
+        // by the following trick, we mark the entry as dirty and commit
+        persistedTasksLicenses.put(license, persistedTasksLicenses.get(license));
+    }
+
+    int getRemainingTokens(String license) {
+        return remainingTokens.get(license);
+    }
+
+    boolean containsLicense(String license) {
+        return remainingTokens.containsKey(license);
+    }
+
+    void setNbTokens(String license, int currentNbTokens) {
+        remainingTokens.put(license, currentNbTokens);
+    }
+
+    public boolean close() {
+        if (recordManager != null) {
+            try {
+                logger.info("Closing Record Manager");
+                recordManager.close();
+                recordManager = null;
+            } catch (IOException e) {
+                logger.warn("Error when closing Record Manager", e);
+            }
+        }
+        return true;
```

Comment:
do we really need to return anything if it is always `true`?

Reply:
Ok.